src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -1219,6 +1227,11 @@ internal TextLoader(IHostEnvironment env, Options options = null, IMultiStreamSo
                 }
             }
 
+            if (options.DecimalMarker != '.' && options.DecimalMarker != ',')
+                throw _host.ExceptUserArg(nameof(Options.DecimalMarker), "Decimal marker cannot be the '{0}' character. It must be '.' or ','.", options.DecimalMarker);
+            if (!options.AllowQuoting && options.DecimalMarker == ',' && options.Separator == ",")
+                throw _host.ExceptUserArg(nameof(Options.AllowQuoting), "Quoting must be allowed if decimal marker and separator are the ',' character.");
```

Comment:
nit: the options.Separator field is only exposed internally through MAML and inside ML.NET's code. In the public API what is actually exposed is options.Separators (plural)... so typiclly options.Separator isn't the option to check in here. Instead you should use _separators.Contains() to check if the separators include , (this is done a couple lines bellow to check the _escapeChar.